providers/openfeature-go-feature-flag-provider/lib/openfeature/go-feature-flag/goff_api.rb

```diff
@@ -4,20 +4,20 @@
 require "net/http"
 require "json"
 require "faraday"
+require "faraday/net_http_persistent"
 require_relative "error/errors"
 require_relative "model/ofrep_api_response"
 
 module OpenFeature
   module GoFeatureFlag
     # This class is the entry point for the GoFeatureFlagProvider
     class GoFeatureFlagApi
-      attr_reader :options
-      def initialize(options: {})
-        @options = options
-        @faraday_connection = Faraday.new(
-          url: @options.endpoint,
-          headers: {"Content-Type" => "application/json"}.merge(@options.custom_headers || {})
-        )
+      def initialize(endpoint: nil, custom_headers: nil)
+        @faraday_connection = Faraday.new(url: endpoint, headers: custom_headers) do |f|
```

Comment:
suggestion(blocking): I think we intend to use the `headers` private you added instead of directly passing custom headers to the connection


```suggestion
        @faraday_connection = Faraday.new(url: endpoint, headers: headers(custom_headers)) do |f|
```